appendix/stata-guide.tex

```diff
@@ -153,151 +198,241 @@ \subsection{Abbreviating commands}
+be used when the information stored is used in a different do-file.
+Globals are error-prone since they are active as long as Stata is open,
+which creates a risk that a global from one project is incorrectly used in another,
+so only use globals where they are necessary.
+Our recommendation is that globals should only be defined in the \textbf{master do-file}.
+All globals should be referenced using both the the dollar sign and curly brackets around their name (\texttt{\$\{\}});
 otherwise, they can cause readability issues when the endpoint of the macro name is unclear.
 
+You should use descriptive names for all macros (up to 32 characters; prefer fewer).
+Simple prefixes are useful and encouraged such as \texttt{thisParam}, \texttt{allParams},
+\texttt{theLastParam}, \texttt{allParams}, or \texttt{nParams}.
 There are several naming conventions you can use for macros with long or multi-word names.
 Which one you use is not as important as whether you and your team are consistent in how you name then.
-You can use all lower case (\texttt{mymacro}), underscores (\texttt{my\_macro}), or ``camel case'' (\texttt{myMacro}), as long as you are consistent.
+You can use all lower case (\texttt{mymacro}), underscores (\texttt{my\_macro}),
+or ``camel case'' (\texttt{myMacro}), as long as you are consistent.
+Nested locals are also possible for a variety of reasons when looping.
+Finally, if you need a macro to hold a literal macro name,
+it can be done using the backslash escape character;
+this causes the stored macro to be evaluated
+at the usage of the macro rather than at its creation.
 
 \codeexample{stata-macros.do}{./code/stata-macros.do}
 
 \subsection{Writing file paths}
 
-All file paths should be absolute and dynamic, should always be enclosed in double quotes, and should
-always use forward slashes for folder hierarchies (\texttt{/}), since Mac and Linux computers cannot read file paths with backslashes.
-File paths should also always include the file extension (\texttt{.dta}, \texttt{.do}, \texttt{.csv}, etc.),
-since to omit the extension causes ambiguity if another file with the same name is created (even if there is a default).
-
-\textbf{Absolute file paths} means that all file paths must start at the root folder of the computer, for
-example \texttt{C:/} on a PC or \texttt{/Users/} on a Mac. This makes sure that
-you always get the correct file in the correct folder. \textbf{We never use \texttt{cd}.}
-We have seen many cases when using \texttt{cd} where
-a file has been overwritten in another project folder where \texttt{cd} was currently pointing to.
-Relative file paths are common in many other programming languages, but there they are relative to the
-location of the file running the code, and then there is no risk that a file is saved in a completely different folder.
+All file paths should always be enclosed in double quotes,
+and should \textbf{always use forward slashes} for folder hierarchies (\texttt{/}).
```

Comment:
```suggestion
and should always use forward slashes for folder hierarchies (\texttt{/}).
```
While your pet peeve, I do not think it warrants bold more than other things here.